fastai/vision/learner.py

```diff
@@ -95,119 +96,114 @@ def unet_learner(data:DataBunch, arch:Callable, pretrained:bool=True, blur_final
+    fig,axes = plt.subplots(rows,rows,figsize=figsize)
+    fig.suptitle('prediction/actual/loss/probability', weight='bold', size=14)
+    for i,idx in enumerate(tl_idx):
+        im,cl = self.data.valid_ds[idx]
+        cl = int(cl)
+        im.show(ax=axes.flat[i], title=
+            f'{classes[self.pred_class[idx]]}/{classes[cl]} / {self.losses[idx]:.2f} / {self.probs[idx][cl]:.2f}')
+
+def plot_multi_top_losses(self, samples:int=3, figsz:Tuple[int,int]=(8,8), save_misclassified:bool=False):
+    "Show images in `top_losses` along with their prediction, actual, loss, and probability of predicted class in a multilabeled dataset."
+    if samples >20:
+        print("Max 20 samples")
+        return
+    losses, idxs = self.top_losses(self.data.c)
+    l_dim = len(losses.size())
+    if l_dim == 1: losses, idxs = self.top_losses()
+    infolist, ordlosses_idxs, mismatches_idxs, mismatches, losses_mismatches, mismatchescontainer = [],[],[],[],[],[]                                                      
+    truthlabels=np.asarray(self.y_true, dtype=int) 
+    classes_ids=[k for k in enumerate(self.data.classes)]
+    predclass=np.asarray(self.pred_class)
+    for i, pred in enumerate(predclass):
+        where_truth=np.nonzero((truthlabels[i]>0))[0]
+        mismatch=np.all(pred!=where_truth)
+        if mismatch: 
+            mismatches_idxs.append(i)
+            if l_dim > 1 : losses_mismatches.append((losses[i][pred],i))
+            else: losses_mismatches.append((losses[i],i))
+        if l_dim > 1: infotup=(i, pred, where_truth, losses[i][pred], np.round(self.probs[i], decimals=3)[pred], mismatch)
+        else: infotup=(i, pred, where_truth, losses[i], np.round(self.probs[i], decimals=3)[pred], mismatch)
+        infolist.append(infotup)
+    ds = self.data.dl(self.ds_type).dataset
+    mismatches = ds[mismatches_idxs]
+    ordlosses=sorted(losses_mismatches, key = lambda x: x[0], reverse=True)
+    for w in ordlosses: ordlosses_idxs.append(w[1])
+    mismatches_ordered_byloss=ds[ordlosses_idxs]
+    print(str(len(mismatches))+' misclassified samples over '+str(len(self.data.valid_ds))+' samples in the validation set.')
+    for ima in range(len(mismatches_ordered_byloss)):
+        mismatchescontainer.append(mismatches_ordered_byloss[ima][0]) 
+    for sampleN in range(samples):
+        actualclasses=''
+        for clas in infolist[ordlosses_idxs[sampleN]][2]:
+            actualclasses=actualclasses+' -- '+str(classes_ids[clas][1])
+        imag=mismatches_ordered_byloss[sampleN][0]
+        imag=show_image(imag, figsize=figsz)
+        imag.set_title(f"""Predicted: {classes_ids[infolist[ordlosses_idxs[sampleN]][1]][1]} \nActual: {actualclasses}\nLoss: {infolist[ordlosses_idxs[sampleN]][3]}\nProbability: {infolist[ordlosses_idxs[sampleN]][4]}""",
+                        loc='left')
+        plt.show()
+        if save_misclassified: return mismatchescontainer
+
+def confusion_matrix(self, slice_size:int=None):
```

Comment:
It's weird to have two implementations of the same thing. There should only be one `confusion_matrix` in the general `ClassificationInterpretation`.